### ISSUE TITLE
[TD-79] Bad error message for wrong hostname configuration

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskserver#80]
_Jens Erat on 2014-10-23T14:32:21Z says:_ 

 When `server` is configured to an invalid hostname, `taskd` exits with the hardly useful log message

{code}
[snip]
2014-10-23 14:20:33 Server starting
2014-10-23 14:20:33 Unknown error
{code}

I wanted to reconfigure from localhost to `0.0.0.0`, but somehow only deleted `local` and saved, resulting in following config:

{code}
server=host:53589
{code}

Hostname `host` is not resolvable.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2014-10-23T14:32:21Z
Modified: 2014-10-24T02:50:14Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2014-10-24T02:50:14Z says:_ 

 Thanks for reporting this!  It was using the wrong error code (errno) when getaddrinfo() returns a code that can be shown by gai_strerror().

Also fixed in Taskwarrior 2.4.0.
